ui/proposal_vote_modal.go

```diff
@@ -0,0 +1,281 @@
+
+type inputVoteOptionsWidgets struct {
+	label      string
+	background color.NRGBA
+	input      decredmaterial.Editor
+	increment  decredmaterial.IconButton
+	decrement  decredmaterial.IconButton
+	max        decredmaterial.Button
+}
+
+type voteModal struct {
+	*pageCommon
+	randomID       string
+	modal          decredmaterial.Modal
+	dialogTitle    string
+	passwordEditor decredmaterial.Editor
+	callback       func(password string, m *voteModal) bool // return true to dismiss dialog
+	btnPositve     decredmaterial.Button
+	btnNegative    decredmaterial.Button
+	yesVote        inputVoteOptionsWidgets
+	noVote         inputVoteOptionsWidgets
+}
+
+func newInputVoteOptions(c *pageCommon, label string) inputVoteOptionsWidgets {
+	i := inputVoteOptionsWidgets{
+		label:      label,
+		background: c.theme.Color.LightGray,
+		input:      c.theme.Editor(new(widget.Editor), ""),
+		increment:  c.theme.PlainIconButton(new(widget.Clickable), c.icons.contentAdd),
+		decrement:  c.theme.PlainIconButton(new(widget.Clickable), c.icons.contentRemove),
+		max:        c.theme.Button(new(widget.Clickable), "MAX"),
+	}
+	i.max.Background = c.theme.Color.Surface
+	i.max.Color = c.theme.Color.Gray2
+	i.max.Font.Weight = text.Bold
+
+	i.increment.Color, i.decrement.Color = c.theme.Color.Text, c.theme.Color.Text
+	i.increment.Size, i.decrement.Size = values.TextSize18, values.TextSize18
+	i.input.Bordered = false
+	i.input.Editor.SetText("0")
+	i.input.Editor.Alignment = text.Middle
+	return i
+}
+
+func newvoteModal(common *pageCommon) *voteModal {
+	cm := &voteModal{
+		pageCommon:  common,
+		randomID:    fmt.Sprintf("%s-%d", ModalInputVote, generateRandomNumber()),
+		modal:       *common.theme.ModalFloatTitle(),
+		btnPositve:  common.theme.Button(new(widget.Clickable), "Vote"),
+		btnNegative: common.theme.Button(new(widget.Clickable), "Cancel"),
+	}
+
+	cm.btnPositve.TextSize, cm.btnNegative.TextSize = values.TextSize16, values.TextSize16
+	cm.btnPositve.Font.Weight, cm.btnNegative.Font.Weight = text.Bold, text.Bold
+	cm.btnPositve.Background = common.theme.Color.Gray1
+	cm.btnPositve.Color = common.theme.Color.Surface
+
+	cm.passwordEditor = common.theme.EditorPassword(new(widget.Editor), "Spending password")
+	cm.passwordEditor.Editor.SingleLine, cm.passwordEditor.Editor.Submit = true, true
+
+	cm.yesVote = newInputVoteOptions(common, "Yes")
+	cm.yesVote.background = common.theme.Color.Success2
+	cm.noVote = newInputVoteOptions(common, "No")
+	return cm
+}
+
+func (cm *voteModal) modalID() string {
+	return cm.randomID
+}
+
+func (cm *voteModal) OnResume() {
+}
+
+func (cm *voteModal) OnDismiss() {
+
+}
+
+func (cm *voteModal) Show() {
+	cm.showModal(cm)
+}
+
+func (cm *voteModal) Dismiss() {
+	cm.dismissModal(cm)
+}
+
+func (cm *voteModal) title(title string) *voteModal {
+	cm.dialogTitle = title
+	return cm
+}
+
+func (cm *voteModal) passwordHint(hint string) *voteModal {
+	cm.passwordEditor.Hint = hint
+	return cm
+}
```

Comment:
These functions can be removed and you can use a default title since this modal is for one purpose.